contents/docs/product-analytics/identify.mdx

```diff
@@ -81,133 +81,106 @@ The ID returned is either the ID automatically generated by PostHog or the ID th
-// You should not use user1 as an alias for user2
-client.alias({
-    distinctId: 'user2',
-    alias: 'user1',
-})
-```
+<h4>Server-side</h4>
 
-```js-web
-// Assume we previously identified a user with 'user1' using posthog.identify('user1')
-// You should not use user1 as an alias for user2
+```node
 client.alias({
-    distinctId: 'user2',
-    alias: 'user1',
+    distinctId: 'distinct_id',
+    alias: 'alias_id',
 })
 ```
 
 ```python
-# Assume we previously identified a user with 'user1' using posthog.identify('user1')
-# You should not use user1 as an alias for user2
-posthog.alias('user2', 'user1');
+posthog.alias(previous_id='distinct_id', distinct_id='alias_id')
 ```
 
 ```php
-# Assume we previously identified a user with 'user1' using posthog.identify('user1')
-# ❌ The following is not possible:
-# You cannot use user1 as an alias for user2
 PostHog::alias(array(
-  'distinctId' => 'user2',
-  'alias' => 'user1'
+  'distinctId' => 'distinct_id',
+  'alias' => 'alias_id'
 ));
 ```
 
 ```ruby
-# Assume we previously identified a user with 'user1' using posthog.identify('user1')
-# You should not use user1 as an alias for user2
-posthog.alias({
-  distinct_id: "user2",
-  alias: "user1",
-})
+posthog.alias(
+  distinct_id: "distinct_id",
+  alias: "alias_id"
+)
 ```
 
 ```go
-// Assume we previously identified a user with 'user1' using posthog.identify('user1')
-// You should not use user1 as an alias for user2
 client.Enqueue(posthog.Alias{
-  DistinctId: "user2",
-  Alias: "user1",
+  DistinctId: "distinct_id",
+  Alias: "alias_id",
 })
 ```
 
 ```java
-// Assume we previously identified a user with 'user1' using posthog.identify('user1')
-// You should not use user1 as an alias for user2
-posthog.alias("user2", "user1");
+posthog.alias("distinct_id", "alias_id");
+```
+
+```bash
+curl -v -L --header "Content-Type: application/json" -d '{
+    "api_key": "<ph_project_api_key>",
+    "properties": {
+        "distinct_id": "distinct_id",
+        "alias": "alias_id"
+    },
+    "timestamp": "2020-08-16 09:03:11.913767",
+    "event": "$create_alias"
+}' <ph_client_api_host>/i/v0/e/
 ```
 
 </MultiLanguage>
 
+The two IDs are merged with the following rules:
+
+- If neither the `distinct_id` nor the `alias_id` are already identified, a new person is created with both IDs.
+- If only one of the IDs is identified, the other ID is added to the existing person.
+- If both IDs are identified:
+  - If both users have merge restrictions (they've been aliased with another user before), the merge is blocked.
+  - If one of the users has merge restrictions and the other does not, the user without merge restrictions is merged into the user with merge restrictions.
+
+This means you can use aliases to merge un-identified, anonymous users into identified users to point anonymous events to an identified person. This also means you can merge users who have been identified before, but have not been aliased with another user.
```

Comment:
Based on this, should we be recommending using `alias` all the time if it covers more cases?

Reply:
This is a good callout. I don't actually like the way I explained this:
- If you Identify on the frontend, it actually does a merge on the anonymous ID + the identified ID. 
- The confusing part: Calling identify on the backend clients doesn't mark someone identified (because the merging of anonymous ID + identified ID is the operation that marks a user identified).
- Calling alias also does anonymous ID + the identified ID merge.

I think this shouldn't be a documentation problem to solve. I am gonna try to get on a call with ingestion and figure out what we want to do. The line between identify and alias is very blurred in this sense.

Reply:
I've always thought of identify on the backend as totally different than identify on the frontend, they have big differences in functionality.